### PR TITLE
[4.x] Allow entries field types to be filtered by `title` as well as `id`

### PR DESCRIPTION
**Pull request by ryanmitchell**
It ended up being a bit more complicated than I thought it would at the outset, but this PR allows you to filter entries fieldtypes by title as well as ID in listing views, which is more intuitive for end users.

The screenshots hopefully explain it better than I just did.

Screenshot 2023-07-18 at 15 48 07

Screenshot 2023-07-18 at 15 48 26

Ideally the same logic could be applied out to other relationship field types.

Closes https://github.com/statamic/cms/issues/8458

### CODE REVIEW

**Comment by jasonvarga** [comment]
Lovely! Ideally we'd have better ways to query relationships, like `whereHas`, but it's not there yet. This is a nice solution until then. 👍

**Comment by ryanmitchell** [comment]
I could work on `whereHas` if you want... :)

File: src/Query/Scopes/Filters/Fields/Entries.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            ->all();
+
+        if (($config['max_items'] ?? 0) == 1) {
+            $query->whereIn($handle, $ids);
+
+            return;
+        }
+
+        if (empty($ids)) {
+            $query->where($handle, -1);
```
Comment (jasonvarga) [inline src/Query/Scopes/Filters/Fields/Entries.php:75]: When `$ids` is empty, I just get an error.
`strtolower(): Argument #1 ($string) must be of type string, array given`

I'm guessing that the `where($handle, -1)` is just trying to trick the query into returning no results?

**Review by jasonvarga — CHANGES_REQUESTED** [review]

File: src/Query/Scopes/Filters/Fields/Entries.php (hunk 1 above)
Comment (ryanmitchell) [reply src/Query/Scopes/Filters/Fields/Entries.php:75]: Yep that was the idea.

Comment (ryanmitchell) [reply src/Query/Scopes/Filters/Fields/Entries.php]: I've tidied that up now - it needed changed to whereJsonContains.

**Review by jasonvarga — APPROVED** [review]

**Review by ryanmitchell — COMMENTED** [review]
I’m not following this change? Should it not be wherein() as we are finding entries where the field matches the ids, not just the first one?

**Comment by jasonvarga** [comment]
Which line are you talking about?

**Comment by ryanmitchell** [comment]
Sorry - was on my phone.  [This change](https://github.com/statamic/cms/pull/8464/commits/d7201077a21cdaed31d3b59f21f172de7b9174a3)

At that point we have an $handle that can have max: 1, but we have $ids that have multiple matches. So whereIn($handle, $ids) was right surely? Otherwise you're only matching on the first $id?

**Comment by jasonvarga** [comment]
> So whereIn($handle, $ids) was right surely?

Yup! 🤦 

CleanShot 2023-08-02 at 16 45 44

**Comment by ryanmitchell** [comment]
Phew, thought I was losing it.

**Comment by jasonvarga** [comment]
Thanks for double checking!

**Pull request merged**